pkg/apis/core/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ExternalEntity struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Desired state of the external entity.
+	Spec ExternalEntitySpec `json:"spec,omitempty"`
+}
+
+// ExternalEntitySpec defines the desired state for ExternalEntity.
+type ExternalEntitySpec struct {
+	// Endpoints is a list of external endpoints associated with this entity.
+	Endpoints []ExternalEndpoint `json:"endpoints,omitempty"`
+	// ExternalNode is the opaque identifier of the agent/controller responsible
+	// for additional computation of this external entity.
+	ExternalNode string `json:"externalNode,omitempty"`
+}
+
+// ExternalEndpoint refers to an endpoint associated with the ExternalEntity.
+type ExternalEndpoint struct {
```

Comment:
I feel this one can be generic but not specific to an ExternalEntity, so we might call just call it Endpoint?

@susanwu88 @abhiraut @tnqn

Reply:
@jianjuns ExternalEndPoint is specific to ExternalEntity today, what do u have in mind in future usage?

Reply:
I do not have a clear idea yet, but as it looks so generic, I assume there can be other API/message to reference it later? For now, it might be relevant to our discussion whether we want to unify Pod and ExternalEntity structs in Controller - Agent messages.

Reply:
@jianjuns. ok sounds good. r there any other concerns with regard to this PR?

Reply:
done.